src/java.base/share/classes/jdk/internal/classfile/impl/AbstractPoolEntry.java

```diff
@@ -544,21 +556,22 @@ public String asInternalName() {
 
     public static final class ClassEntryImpl extends AbstractNamedEntry implements ClassEntry {
 
-        public ClassDesc sym = null;
+        public @Stable ClassDesc sym;
```

Comment:
As [GH‑20957] has now been merged, this can be made to use the `Utf8Entry​.typeSym` field, which would also speed up [`ConstantPoolBuilder​::classEntry​(Utf8Entry)`].

[GH‑20957]: https://github.com/openjdk/jdk/pull/20957
[`ConstantPoolBuilder​::classEntry​(Utf8Entry)`]: https://docs.oracle.com/en/java/javase/22/docs/api/java.base/java/lang/classfile/constantpool/ConstantPoolBuilder.html#classEntry(java.lang.classfile.constantpool.Utf8Entry)

Reply:
I prefer not to use that field for internal name utf8 entries; that is reserved for descriptor matches, so I reuse only for arrays.